addons/website/models/website.py

```diff
@@ -675,8 +675,98 @@ def configurator_apply(self, **kwargs):
         IrQweb = self.env['ir.qweb'].with_context(website_id=website.id, lang=website.default_lang_id.code)
         snippets_cache = {}
         translated_content = {}
+        fonts_attributes = {}
+        html_storage = {}
+
+        def _replace_tags_with_placeholders(html_string):
+            parser = html.HTMLParser()
+            tree = html.fromstring(f'<div>{html_string}</div>', parser=parser)
+
+            def _construct_html(elements):
+                opening_tags = []
+                closing_tags = []
+
+                for element in elements[1:]:
+                    tag = element['tag']
+                    attr = element['attr']
+
+                    if attr:
+                        attr_string = ' '.join([f'{key}="{value}"' for key, value in attr.items()])
+                        opening_tag = f'<{tag} {attr_string}>'
+                    else:
+                        opening_tag = f'<{tag}>'
+
+                    closing_tag = f'</{tag}>'
+
+                    opening_tags.append(opening_tag)
+                    closing_tags.append(closing_tag)
+
+                # Combine all opening tags, add placeholder $0, and close all tags
+                html_string = ''.join(opening_tags) + '$0' + ''.join(closing_tags[::-1])
+
+                return html_string
+
+            def _generate_unique_hash(element_text, element_tag, element_attrib):
+                attrib_string = ','.join(f'{key}={value}' for key, value in sorted(element_attrib.items()))
+                combined_string = f'{element_text}|{element_tag}|{attrib_string}'
+                hash_object = hashlib.sha256(combined_string.encode('utf-8'))
+                return hash_object.hexdigest()
+
+            # Function to replace specific tags with placeholders
+            def _process_element(element):
+                # Handle highlights SVG.
+                if element.tag == 'span' and len(element) == 1 and element[0].tag == 'span' and len(element[0]) == 1 and element[0][0].tag == 'svg':
```

Comment:
A saved highlighted text is just a unique span, no svg inside.
E.g.
```xml
<span class="o_text_highlight o_translate_inline o_text_highlight_scribble_4 o_text_highlight_fill" style="--text-highlight-width: 3px;">Enhance</span>
```
The SVG is constructed on page load.

Did we do something different in some snippet / theme templates? Is not that an issue? cc @bvr-odoo @bso-odoo Maybe it's something about the page dialog / snippet dialog previews I don't know about?

Reply:
Here, the "svg" is hardcoded -> https://github.com/odoo/design-themes/commit/c31b3c7c8327b4d8c3924de4bfca9ca2db26148b#diff-f2c8ab0f7de73b89757170a6672b0a68e88e99a1d06747db839431da9333bb19R6-R8

Reply:
@xO-Tx What do you think about it?

Reply:
I've tested and it works without the svg tag.
![image](https://github.com/user-attachments/assets/7597f059-b39c-48a4-bc99-1f3729a299b9)
```xml
<span class="o_text_highlight o_translate_inline o_text_highlight_circle_2 o_text_highlight_fill" style="--text-highlight-width: 6px;">potential</span>
```

Reply:
@qsm-odoo The "New Page Dialog" and the block dialog preview do not show them if the SVG is not there.
But the created page does contain it in the HTML editor: it should be cleaned upon selection. (Or we should make the preview systems handle the highlight display and remove the SVGs from themes)

Reply:
The public widget will apply every detected highlight effect on the page (that should be saved in the minimal format) and [won't reapply the effect if the highlight is already built](https://github.com/odoo/odoo/blob/master/addons/website/static/src/js/text_processing.js#L292-L295).

For the "New Page Templates" & D&D preview, it's normal that we don't have the effects since the assets are not available in the `o_add_snippets_preview iframe`... @bvr-odoo isn't possible to add the assets to the iframe with something similar to what we do in `AddSnippetDialog.insertStyle()`  :thinking:?

Remarks:

1. @SergeBayet I didn't check the diff here in detail, but it seems that it only handles `element[0]` (= the first `<span class="o_text_highlight_item".../>`). A highlighted text has the following format:
```xml
<span class="o_text_highlight">
   <span class="o_text_highlight_item">
       line1-textNode1 [line1-textNode2,...]
       <svg.../>
   </span>
   [<br/>]
   <span class="o_text_highlight_item">
       line2-textNode1 [line2-textNode2,...]
       <svg.../>
   </span>
   ...
</span>
```
Which means the code here will only handle "one line" highlights :grimacing: 

2. Also, adding the highlights explicitly in themes should take into consideration the content width. If the text is automatically split because of the width of its container, the highlight won't be adapted correctly without the JS code :grimacing:

[master.rev-highlights.webm](https://github.com/user-attachments/assets/827699ed-3b1c-430b-8185-4e3a0c268567)

Reply:
> isn't possible to add the assets to the iframe with something similar to what we do in AddSnippetDialog.insertStyle() 🤔?

@xO-Tx JS is not loaded into the iframe of the "snippets modal" but as BSO said, it should be possible to use the preview system by adding the `s_dialog_preview` class to the SVG tag. I don't think a similar system exists for the new page template, so it may be necessary to implement it there if they are page previews with highlighted texts to display.

Reply:
`s_dialog_preview` will not work out-of-the-box to handle this at all: it's not just about removing the SVG but also the wrapper which contains the text... which you don't want to remove.

What the visitor has
```xml
<h1>
    Unleash your
    <span class="o_text_highlight o_translate_inline o_text_highlight_circle_2 o_text_highlight_fill" style="--text-highlight-width: 6px;">
        <span class="o_text_highlight_item">
            potential
            <svg><!-- much stuff !--></svg>
        </span>
    </span>
</h1>
```
What is saved in edit mode
```xml
<h1>
    Unleash your
    <span class="o_text_highlight o_translate_inline o_text_highlight_circle_2 o_text_highlight_fill" style="--text-highlight-width: 6px;">
        potential
    </span>
</h1>
```

=> you cannot just put s_dialog_preview somewhere: on the svg, you don't remove enough and on the inner span, you remove too much.

It's important we solve this soon, I'd say:
1. Someone make a PR converting all the snippets using the **saved** form mentioned above (just the `o_text_highlight` span, without the inner svg and `o_text_highlight_item` spans.
2. @SergeBayet can thus rely on that simple `span.o_text_highlight` element here
3. We make sure the snippet dialog and add page dialogs properly show text highlight. I don't think there is a need to make the JS and the related public widget available in the iframe at all. Can't the dialog code itself use the utils @xO-Tx made (`applyTextHighlight`) from outside the iframes to just apply it on all in-the-iframe `.o_text_highlight` ? We then don't need to care about what the public widget does (checking resizes, destroying, etc), it's only for the duration of the preview in the dialog anyway?

What do you think?

Reply:
@qsm-odoo I'll start on this right away between theme review iterations.

Reply:
Thank you :pray: